### DISCUSSION TITLE
Manual trigger of scheduled chains retrieval

### DISCUSSION THREAD

**Discussion by cr1xu5** [cybertec-postgresql/pg_timetable#592] [Q&A] [answered]
Thank you for pg_timetable!

**Is your feature request related to a problem? Please describe.**
When using pg_timetable in an integration test, the time until the next "Retrieve scheduled chains to run" is unknown and this makes the integration tests flaky.

**Describe the solution you'd like**
For example an API endpoint that triggers the retrieval of scheduled chains to run.

**Describe alternatives you've considered**
- Put various `sleep` moments in the integration tests.
- Stop testing with pg_timetable in the integration tests and test pg_timetable outside of CI/CD.

**Comment by pashagolub** [comment, marked as answer]
Hello.

Have you considered starting chains asynchronously with `timetable.notify_chain_start()` or with [REST API endpoint](https://pg-timetable.readthedocs.io/en/master/api.html#chain-management-endpoints)?

If you're building sources yourself, you can change [`refetchTimeout` constant](https://github.com/cybertec-postgresql/pg_timetable/blob/7d64d34de150986c8a33da194d75ee4d68447e56/internal/scheduler/scheduler.go#L14) for tests.
